compiler/rustc_codegen_ssa/src/traits/builder.rs

```diff
@@ -202,6 +205,177 @@ pub trait BuilderMethods<'a, 'tcx>:
     fn intcast(&mut self, val: Self::Value, dest_ty: Self::Type, is_signed: bool) -> Self::Value;
     fn pointercast(&mut self, val: Self::Value, dest_ty: Self::Type) -> Self::Value;
 
+    fn cast_float_to_int(
+        &mut self,
+        signed: bool,
+        x: Self::Value,
+        dest_ty: Self::Type,
+    ) -> Self::Value {
+        let in_ty = self.cx().val_ty(x);
+        let (float_ty, int_ty) = if self.cx().type_kind(dest_ty) == TypeKind::Vector
+            && self.cx().type_kind(in_ty) == TypeKind::Vector
+        {
+            (self.cx().element_type(in_ty), self.cx().element_type(dest_ty))
+        } else {
+            (in_ty, dest_ty)
+        };
+        assert!(matches!(self.cx().type_kind(float_ty), TypeKind::Float | TypeKind::Double));
+        assert_eq!(self.cx().type_kind(int_ty), TypeKind::Integer);
```

Comment:
Hm, isn't this redundant with the checks performed by calling `float_width` and then matching on that? I suppose it doesn't exactly hurt to try to catch it in a single place here but...

Reply:
Yeah, it should error there, but I figure it's good to ensure that everything is the way we expect before going on.